isis/src/base/apps/topds4/topds4.cpp

```diff
@@ -58,13 +60,41 @@ namespace Isis {
         // get the xml label and add it to the template data
       }
     }
-    
+
     std::cout << "=================================" << std::endl;
     std::cout << dataSource.dump(4) << std::endl;
     std::cout << "=================================" << std::endl;
 
-    Environment env;
     std::string inputTemplate = ui.GetFileName("TEMPLATE").toStdString();
+
+    Environment env;
+    // Call back functions
+    /**
+     * Renders to the current UTC time formatted as YYYY-MM-DDTHH:MM:SS
+     */
+    env.add_callback("currentTime", 0, [](Arguments& args) {
+      time_t startTime = time(NULL);
+      struct tm *tmbuf = gmtime(&startTime);
+      char timestr[80];
+      strftime(timestr, 80, "%Y-%m-%dT%H:%M:%S", tmbuf);
+      return string(timestr);
+    });
+
+    /**
+     * Renders to the filename of the output img
+     */
+    env.add_callback("imageFileName", 0, [icube](Arguments& args) {
+      QString cubeFilename = icube->fileName().split("/").back();
+      return (cubeFilename.split(".")[0] + ".img").toStdString();
+    });
+
+    /**
+     * Renders to the MD5 hash for the input cube
+     */
+    env.add_callback("md5Hash", 0, [icube](Arguments& args) {
+      md5wrapper md5;
+      return md5.getHashFromFile(icube->fileName()).toStdString();
```

Comment:
Is this for the case where a pds4 label points to an ISIS cube? If so, then should we change the name to reflect that? I'm thinking there may be other md4Hash functions (e.g., The xml may point to a DN only file). If not then when would we use this specific callback?

Reply:
I just put this in to see if we could call stuff from other places in the callbacks. This comes from lronac2pds where it is the checksum of the output file. Once we are properly exporting the output file then I'll change it to that.